test/integration/controlplane/transformation/kms_transformation_test.go

```diff
@@ -536,6 +539,91 @@ resources:
 	}
 }
 
+func TestEncryptAll(t *testing.T) {
+	testCases := []struct {
+		name             string
+		encryptionConfig string
+	}{
+		{
+			name: "encrypt all resources",
+			encryptionConfig: `
+kind: EncryptionConfiguration
+apiVersion: apiserver.config.k8s.io/v1
+resources:
+  - resources:
+    - '*.*'
+    providers:
+    - kms:
+        name: encrypt-all-kms-provider
+        cachesize: 1000
+        endpoint: unix:///@encrypt-all-kms-provider.sock		
+`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pluginMock, err := mock.NewBase64Plugin("@encrypt-all-kms-provider.sock")
+			if err != nil {
+				t.Fatalf("failed to create mock of KMS Plugin: %v", err)
+			}
+
+			go pluginMock.Start()
+			if err := mock.WaitForBase64PluginToBeUp(pluginMock); err != nil {
+				t.Fatalf("Failed start plugin, err: %v", err)
+			}
+			defer pluginMock.CleanUp()
+
+			test, err := newTransformTest(t, tc.encryptionConfig, false, "")
+			if err != nil {
+				t.Fatalf("failed to start KUBE API Server with encryptionConfig")
+			}
+			defer test.cleanUp()
+
+			_, serverResources, err := test.restClient.Discovery().ServerGroupsAndResources()
+			if err != nil {
+				t.Fatal(err)
+			}
+			resources := etcd.GetResources(t, serverResources)
+
+			wantPrefix := "k8s:enc:kms:v1:encrypt-all-kms-provider:"
+			etcdStorageData := etcd.GetEtcdStorageDataForNamespace(testNamespace)
+			for _, resource := range resources {
+				gvr := resource.Mapping.Resource
+				stub := etcdStorageData[gvr].Stub
+
+				dynamicClient, obj, err := etcd.JSONToUnstructured(stub, testNamespace, &meta.RESTMapping{
+					Resource:         gvr,
+					GroupVersionKind: gvr.GroupVersion().WithKind(resource.Mapping.GroupVersionKind.Kind),
+					Scope:            resource.Mapping.Scope,
+				}, dynamic.NewForConfigOrDie(test.kubeAPIServer.ClientConfig))
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				_, err = dynamicClient.Create(context.Background(), obj, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				response, err := test.readRawRecordFromETCD(
+					fmt.Sprintf("/%s%s",
+						strings.Split(test.storageConfig.Prefix, "/")[0],
+						etcdStorageData[gvr].ExpectedEtcdPath,
+					),
+				)
+				if err != nil {
+					t.Fatalf("failed to read from etcd: %v", err)
+				}
+
+				if !bytes.HasPrefix(response.Kvs[0].Value, []byte(wantPrefix)) {
+					t.Fatalf("expected resource %s to be prefixed with %s, but got %s", gvr.Resource, wantPrefix, response.Kvs[0].Value)
+				}
```

Comment:
Do not read one record at a time at a specific path.  Read from the root (`test.storageConfig.Prefix`) recursively via the etcd client and assert everything is encrypted (this would be done outside of the loop).